src/controllers/authenticationController.ts

```diff
@@ -130,13 +134,17 @@ export default class AuthenticationController {
     | IResponseErrorNotFound
     | IResponseSuccessJson<PublicSession>
   > {
+    // This controller doesn't have a Passport middleware that checks the Bearer token, we need to check it here.
     const maybeToken = await this.extractTokenFromRequest(req);
     if (isNone(maybeToken)) {
       return ResponseErrorInternal("No token in the request");
     }
 
+    // The token is present, look for a session.
     const sessionToken = maybeToken.value;
-    const errorOrSession = await this.sessionStorage.get(sessionToken);
+    const errorOrSession = await this.sessionStorage.get(
+      sessionToken as SessionToken
```

Comment:
the point here is that `extractTokenFromRequest` should return a `SessionToken` so you don't have to cast it later (basically `extractTokenFromRequest` "decodes" the raw string into a `SessionToken`)